forc-util/src/lib.rs

```diff
@@ -64,6 +64,26 @@ pub fn format_log_receipts(receipts: &[fuel_tx::Receipt], pretty_print: bool) ->
     }
 }
 
+/// Continually go down in the file tree until a Forc manifest file is found.
+///
+/// Given a pacakge's directory (which contains a manifest file) goes down in the file tree until
+/// another manifest file is found.
+///
+/// Returns the directory that contains the nested manifest file.
+pub fn find_nested_manifest_dir(starter_path: &Path) -> Option<PathBuf> {
+    use walkdir::WalkDir;
+    WalkDir::new(starter_path)
+        .into_iter()
+        .filter_map(|e| e.ok())
+        .filter(|entry| entry.file_name().to_string_lossy() == constants::MANIFEST_FILE_NAME)
+        .map(|entry| {
+            let mut entry = entry.path().to_path_buf();
+            entry.pop();
+            entry
+        })
+        .nth(1)
+}
```

Comment:
Perhaps we should split this into `find_nested_dir_with_file` and `find_nested_manifest_dir` like we do for the parent traversal functions?